attempt a better README (#66)
nevinera/quiet_quality: README.md

# QuietQuality

> There are a lot of different tools that you need to run as you work - possibly before you commit,

Reviewer: MarkdownLint wants a space between headers and text:

https://github.com/DavidAnson/markdownlint/blob/v0.28.2/doc/md022.md
```suggestion

There are a lot of different tools that you need to run as you work - possibly before you commit,
```

Author: I'd rather use a ruby markdown linter, but `mdl` doesn't have the same rules. Is `markdownlint` the more typical tool?

Author: (Whichever we want to use, I'll ticket to add tool support for it to QQ, so I'd rather choose the more heavily-used option)

Author: the ruby and node versions have 1.5k and 3.8k stars relatively, way ahead of the remaining options. I might add support for both tools? But I think I'd rather use the ruby one in this repo.

Reviewer: Let me check what my editor is using - I'm seeing intellisense markers inline :)

Reviewer: Seems like the vscode plugin _I chose_ is using https://github.com/DavidAnson/markdownlint

Author: they do actually seem to have very similar rules, but wildly different systems for configuration :-)
I went with the ruby gem `mdl`, but haven't implemented actual tool support in QQ yet. Doesn't look hard though, I think I'll tackle a few more tools after the current suite of "make it more pleasant to use" improvements around logging output.

but you want to avoid introducing _new_ issues, without having to first resolve all of the existing
ones. QuietQuality can help with that too.

## Tool Support
So far, we have support for the following tools:

* rubocop
* standardrb
* rspec
* haml-lint
* brakeman (though there's no way to run this against only changed files)

Supporting more tools is relatively straightforward - they're implemented by wrapping cli
invocations and parsing output files (which overall seem to be much more stable interfaces
than the code interfaces to the various tools), and each tool's support is built orthogonally
to the others, in a `QuietQuality::Tools::[Something]` namespace, with a `Runner` and a `Parser`.

## Local Usage Examples
Working locally, you'll generally want to commit a `.quiet_quality.yml` configuration file into
the root of your repository - it'll specify which tools to run by default, and how to run them
(whether you want to only run each tool against the "changed files", whether to filter the
resulting messages down to only those targeting lines that have been changed), and allows you to

> specify the 'comparison branch', so you don't have to make a request to your origin server every

Reviewer: ```suggestion
specify the _comparison branch_', so you don't have to make a request to your origin server every
```

Author: Why are we italicizing here? I was using quotes because it was an imprecise phrase (as it's either `--comparison-branch` or `comparison_branch` depending on whether we're in the cli or a configuration file). If we'd like to settle on italics for option names in general, I think that'd be fine.

Author: Ah, I think this is connected to the comment below it.

> time you run the tool to see whether you're comparing against `master` or `main` in this project.

Reviewer: Maybe try for more consistency (and visual differentiation) for the behaviors that map to parameter options?

```suggestion
the root of your repository - it'll specify which tools to run by default, and how to run them
(whether you want to only run each tool against the _changed files_, whether to _filter_ the
resulting _messages_ down to only those targeting lines that have been changed), and allows you to
specify the _comparison branch_, so you don't have to make a request to your origin server every
time you run the tool to see whether you're comparing against `master` or `main` in this project.
```
